src/utilities/modules/temperatureSensorHelpers.js

```diff
@@ -370,87 +378,119 @@ function addHeadAndTrailingLogToBreach({ result, sensor, database }) {
+  return [
+    createGenericSensorLog({
+      ...meanSensorLog,
       aggregation: 'aggregate',
-    });
-
-    lastAggregationTo = aggregateTo;
-    aggregateFrom = aggregateTo;
-    aggregateTo = new Date(aggregateFrom.getTime() + fullAggregateInterval);
-    aggregateMiddlePoint = new Date(aggregateFrom.getTime() + halfFullAggregateInterval);
-  }
-  if (lastAggregationTo == null) {
-    if (lastFullAggregation !== null) lastAggregationTo = lastFullAggregation;
-    else lastAggregationTo = new Date(2019, 1, 1);
-  }
-
-  const logsToDelete = sensorLogs
-    .filtered('aggregation == "preAggregate" && timestamp < $0', lastAggregationTo)
-    .slice();
-  const fullAggregationDeletions = logsToDelete.length;
-  database.write(() => {
-    logsToDelete.forEach(sensorLog => {
-      database.delete('SensorLog', sensorLog);
-    });
-  });
+      temperature: Math.max(...sensorLogGroup.map(({ max }) => max)),
+    }),
+    createGenericSensorLog({
+      ...meanSensorLog,
+      temperature: Math.min(...sensorLogGroup.map(({ min }) => min)),
+    }),
+  ];
+}
 
-  database.write(() => {
-    logsToAdd.forEach(sensorLog => {
-      const newSensorLog = database.update('SensorLog', sensorLog);
-      sensor.sensorLogs.push(newSensorLog);
-      linkToBatches(newSensorLog, database);
-    });
+/**
+ * Utility method used to aggregate pre-aggregate sensor logs for a given
+ * sensor.
+ *
+ * Pre-aggregate sensor logs: sensor log for a 20 minute time interval.
+ * Aggregate sensor log: aggregated sensor log for a 8 hour interval.
+ *
+ * Takes groups of pre-aggregate sensor logs and creates two new
+ * sensor logs for each group - each having a temperature equal
+ * to the max and min temperature within the group of sensor logs.
+ * The timestamp is equal to the median timestamp of all sensor
+ * logs within the group.
+ *
+ * Each group to be aggregated is a collection of sequential sensor logs which fit
+ * within an 8 hour interval beginning with the first sensor logs timestamp. Each
+ * pre-aggregate sensor log is for a 20 minute interval, so there will be a
+ * maximum of 24 sensor logs for each full aggregate log.
+ * If within the 8 hour period a breach occurred, there will be less sensorlogs,
+ * to a minimum of 1.
+ * @param {object}       result   object containing the result of aggregation
+ * @param {Realm.Sensor} sensor   The Sensor object for which the logs should be aggregated
+ * @param {Realm}        database App-wide database entry point
+ */
+function doFullAggregation({ result, sensor, database }) {
+  const { sensorLogs } = sensor;
+  if (!sensorLogs || sensorLogs.length === 0) return null;
+  // Want to leave the last 2 hours of logs un-aggregated. Filter them
+  // out.
+  const maxTimeToAggregate =
+    sensorLogs.filtered('aggregate == $0', 'preAggregate').max('timestamp') -
+    TWO_HOURS_MILLISECONDS;
+  const sortedSensorLogs = sensorLogs.filtered(
```

Comment:
Can you get this sorted

Reply:
oops 3a18cca9ff6858e75629009e3d9d62776b07319d